__tests__/app.test.js

```diff
@@ -61,6 +61,75 @@ describe("/api/articles", () => {
           expect(articles).toBeSortedBy("created_at", { descending: true });
         });
     });
+    describe("topic query", () => {
+      test("should GET 200: response filters by topic when topic query given", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch")
+          .expect(200)
+          .then((res) => {
+            const { articles } = res.body;
+            expect(articles.length).toBe(11);
+            articles.forEach((article) => {
+              expect(article).toMatchObject({
+                topic: "mitch",
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                comment_count: expect.any(String),
+              });
+            });
+          });
+      });
+      test("should GET 200: responds with empty array if topic that doesnt exist passed", () => {
+        return request(app)
+          .get("/api/articles?topic=hello")
+          .expect(200)
+          .then((res) => {
+            const { articles } = res.body;
+            expect(articles.length).toBe(0);
```

Comment:
We actually want a 404 for topic queries that are non-existant - if a topic exists but has no articles, that's when we want to give a 200 + empty array